### PR TITLE
Add more numeric matchers

### PR DESCRIPTION
**Pull request by natebosch**
- Add `<=` and `>=` to complement `<` and `>`.
- Add `isNaN` and `isNotNaN`.
- Add `closeTo`.
- Add tests for the new conditions.
- Use lowercase for the `which` on `<` and `>`.
- Add positive and negative conditions for `isNegative`, `isFinite`, and `isInfinite`.
- Split the test for less than and greater than into separate cases and
  add an equal case.

### CODE REVIEW

File: pkgs/checks/lib/src/extensions/math.dart [hunk 1]
```diff
@@ -5,21 +5,128 @@
...
       return Rejection(
           actual: literal(actual),
-          which: ['Is not greater than ${literal(other)}']);
+          which: ['is not greater than ${literal(other)}']);
     });
   }
 
+  /// Expects that this number is greater than or equal to [other].
+  void operator >=(num other) {
+    context.expect(() => ['is greater than ${literal(other)}'], (actual) {
```
Comment (jakemac53) [inline pkgs/checks/lib/src/extensions/math.dart:20]: nit: greater than or equal to (same below)

File: pkgs/checks/lib/src/extensions/math.dart [hunk 2]
```diff
@@ -5,21 +5,128 @@
...
+      return Rejection(actual: literal(actual), which: ['is not finite']);
+    });
+  }
+
+  /// Expects that [num.isFinite] is false.
+  ///
+  /// Satisfied by [double.nan], [double.infinity] and
+  /// [double.negativeInfinity].
+  void isNotFinite() {
+    context.expect(() => ['is finite'], (actual) {
```
Comment (jakemac53) [inline pkgs/checks/lib/src/extensions/math.dart:94]:
```suggestion
    context.expect(() => ['is not finite'], (actual) {
```

File: pkgs/checks/lib/src/extensions/math.dart [hunk 3]
```diff
@@ -5,21 +5,128 @@
...
+      if (actual.isInfinite) return null;
+      return Rejection(actual: literal(actual), which: ['is not infinite']);
+    });
+  }
+
+  /// Expects that [num.isInfinite] is false.
+  ///
+  /// Satisfied by [double.nan] and finite numbers.
+  void isNotInfinite() {
+    context.expect(() => ['is infinite'], (actual) {
```
Comment (jakemac53) [inline pkgs/checks/lib/src/extensions/math.dart:114]:
```suggestion
    context.expect(() => ['is not infinite'], (actual) {
```

File: pkgs/checks/test/extensions/math_test.dart [hunk 4]
```diff
@@ -9,16 +9,205 @@ import 'package:test/scaffolding.dart';
...
+        checkThat(softCheck<int>(42, (p0) => p0 < 42))
+            .isARejection(actual: '<42>', which: ['is not less than <42>']);
+      });
+    });
+
+    group('less than or equal', () {
+      test('succeeds for happy case', () {
+        checkThat(42) <= 50;
+      });
+      test('fails for less than', () {
```
Comment (jakemac53) [inline pkgs/checks/test/extensions/math_test.dart:58]:
```suggestion
      test('fails for greater than', () {
```

File: pkgs/checks/test/extensions/math_test.dart [hunk 5]
```diff
@@ -9,16 +9,205 @@ import 'package:test/scaffolding.dart';
...
+        checkThat(softCheck<int>(42, (p0) => p0 <= 7)).isARejection(
+            actual: '<42>', which: ['is not less than or equal to <7>']);
+      });
+      test('succeeds for equal', () {
+        checkThat(42) <= 42;
+      });
+    });
+
+    group('isNan', () {
```
Comment (jakemac53) [inline pkgs/checks/test/extensions/math_test.dart:67]:
```suggestion
    group('isNaN', () {
```

File: pkgs/checks/test/extensions/math_test.dart [hunk 6]
```diff
@@ -9,16 +9,205 @@ import 'package:test/scaffolding.dart';
...
+        checkThat(softCheck<num>(42, (c) => c.isNaN()))
+            .isARejection(actual: '<42>', which: ['is a number']);
+      });
+      test('fails for numeric doubles', () {
+        checkThat(softCheck<num>(42.1, (c) => c.isNaN()))
+            .isARejection(actual: '<42.1>', which: ['is a number']);
+      });
+    });
+
+    group('isNan', () {
```
Comment (jakemac53) [inline pkgs/checks/test/extensions/math_test.dart:81]:
```suggestion
    group('isNotNan', () {
```

File: pkgs/checks/lib/src/extensions/math.dart (hunk 1 above)
Comment (natebosch) [reply pkgs/checks/lib/src/extensions/math.dart:20]: Done.

**Review by jakemac53 — APPROVED** [review]

**Pull request merged**